latex/epoch.tex

```diff
@@ -1,54 +1,614 @@
+  \label{fig:rules:utxoep}
+\end{figure}
 
 %%
-%% Figure - Pool Rules
+%% Figure - Accounting Defs
 %%
 \begin{figure}
+  \emph{Accounting Fields}
+  \begin{equation*}
+    \Accnt =
+    \left(
+      \begin{array}{r@{~\in~}lr}
+        \var{treasury} & \Coin & \text{treasury pool}\\
+        \var{reserves} & \Coin & \text{reserve pool}\\
+        \var{rewardPool} & \Coin & \text{reward pool}\\
+      \end{array}
+    \right)
+  \end{equation*}
+  %
+  \emph{Accounting environment}
+  \begin{equation*}
+    \AccntEnv =
+    \left(
+      \begin{array}{r@{~\in~}lr}
+        \var{slot} & \Slot & \text{current slot}\\
+        \var{pc} & \PrtclConsts & \text{protocol constants}\\
+        \var{production} & \Production & \text{blocks produced in the epoch}\\
+        \var{utxoSt} & \UTxOState & \text{utxo state}\\
+        \var{pstate} & \PState & \text{pool state}\\
+      \end{array}
+    \right)
+  \end{equation*}
+  %
+  \emph{Accounting States}
+  \begin{equation*}
+    \AccntState =
+    \left(
+      \begin{array}{r@{~\in~}lr}
+        \var{accnt} & \Accnt & \text{accounting}\\
+        \var{dstate} & \DState & \text{delegation state}\\
+      \end{array}
+    \right)
+  \end{equation*}
+  %
+  \emph{Accounting transitions}
+  \begin{equation*}
+    \_ \vdash
+    \var{\_} \trans{accnt}{} \var{\_}
+    \subseteq \powerset (\AccntEnv \times \AccntState \times \AccntState)
+  \end{equation*}
+  %
+  \caption{Accounting transition-system types}
+  \label{fig:ts-types:accnt}
+\end{figure}
 
-  \begin{equation}\label{eq:pool-reap}
-    \inference[Pool-Reap]
+%%
+%% Figure - Accounting Rules
+%%
+\begin{figure}
+  \begin{equation}\label{eq:accnt}
+    \inference[Accounting]
+    {
+      \var{obl} = \obligation{pc}{stkeys}{stpools}{slot} \\
+      \var{decayed} = \var{deposits} - \var{obl} \\
+      \rho, \tau \in pc \\
+      \var{expansion} = \floor*{\rho \cdot \var{reserves}} \\
+      \var{totalPool} = \var{fees} + \var{decayed} + \var{rewardPool} + \var{expansion} \\
+      \var{newTreasury} = \floor*{\tau \cdot \var{totalPool}} \\
+      \var{availablePool} = \var{totalPool} - \var{newTreasury} \\
+      \var{rewards'} = \reward{production}{pc}{availablePool}{dwstate}{rewards}\\
+      \var{paidRewards} = \sum\limits_{\_\mapsto c\in\var{rewards'}}c
+    }
+    {
+      \begin{array}{l}
+        \var{production}\\
+        \var{slot}\\
+        \var{pc}\\
+        \var{utxoSt}\\
+        \var{pstate}\\
+      \end{array}
+      \vdash
+      \left(
+        \begin{array}{r}
+          \var{treasury} \\
+          \var{reserves} \\
+          \var{rewardPool} \\
+          \var{stkeys} \\
+          \var{rewards} \\
+          \var{delegations}
+        \end{array}
+      \right)
+      \trans{accnt}{}
+      \left(
+        \begin{array}{rcl}
+          \var{treasury} & + & \var{newTreasury}\\
+          \var{reserves} & - & \var{expansion} \\
+          \var{availablePool} & - & \var{paidRewards} \\
+          \var{stkeys} \\
```

Comment:
If `stkeys` and `delegations` don't change we might as well remove them from the `ACCNT` state.

Reply:
Damanian and I chatted about this over a call.  It's true that those values do not change, but it is written this way in order to pack the data in a way for the big `EPOCH` rule to stay readable.  Simplifications would be great if we can find them...